rustc_codegen_spirv/src/builder/builder_methods.rs

```diff
@@ -116,6 +104,40 @@ fn memset_dynamic_scalar(
 }
 
 impl<'a, 'tcx> Builder<'a, 'tcx> {
+    fn ordering_to_semantics_def(&self, ordering: AtomicOrdering) -> SpirvValue {
+        let mut invalid_seq_cst = false;
+        let semantics = match ordering {
+            AtomicOrdering::NotAtomic => MemorySemantics::NONE,
+            AtomicOrdering::Unordered => MemorySemantics::NONE,
+            AtomicOrdering::Monotonic => MemorySemantics::NONE,
+            AtomicOrdering::Acquire => MemorySemantics::MAKE_VISIBLE | MemorySemantics::ACQUIRE,
+            AtomicOrdering::Release => MemorySemantics::MAKE_AVAILABLE | MemorySemantics::RELEASE,
+            AtomicOrdering::AcquireRelease => {
+                MemorySemantics::MAKE_AVAILABLE
+                    | MemorySemantics::MAKE_VISIBLE
+                    | MemorySemantics::ACQUIRE_RELEASE
+            }
+            AtomicOrdering::SequentiallyConsistent => {
+                let emit = self.emit();
+                let memory_model = emit.module_ref().memory_model.as_ref().unwrap();
+                if memory_model.operands[1].unwrap_memory_model() == MemoryModel::Vulkan {
+                    invalid_seq_cst = true;
```

Comment:
It is simpler to directly return the zombie here I think.

Reply:
eh, there's some weird RefCell already borrowed shenanigans going on if structured that way - the `memory_model` line takes a ref to the module, and then creating the zombie takes another borrow, so that means the borrow needs to be dropped before that, and it turns into a weird fragile mess.

(really dislike that codegen_ssa requires RefCells everywhere, but, oh well...)